### PR TITLE
6387 update_kwargs for merging multiple configs

### LINKED ISSUE #5899: Merging/overriding configs with same format in monai bundle run
**Is your feature request related to a problem? Please describe.**
Currently, overriding config can be done through an `args_file` argument pointing to a json/yaml provided to the `run` script. However, it only merges configs at the base level, and to override nested structures, the `#` format must be used. 

Example,
````
monai.bundle.run("", **{
    "args_file": "./config2.json",
    "config_file": "./config1.json", 
    
})
````

````
```
config1.json
```
{
    "trainer":
    {
        "max_epochs": 5

    }
}
````

````
```
config2.json
```
{
    "trainer#devices": 1
}
````

This gives the output,
````
[{'_meta_': {}, 'trainer': {'max_epochs': 5, 'devices': 1}}]
````

However, if the config provided to  `args_file` is in the same format (shown below) as the `config_file`, it does not do the merge,
````
```
config2.json
```
{
    "trainer":
    {
        "devices": 1

    }
}
````
**Describe the solution you'd like**
It would be beneficial to have the ability to merge these two configs when they are in the same format to allow for use cases where a default config is set to have good parameter definitions, and the user only needs to define/override specific fields.

**Describe alternatives you've considered**
I tried specifying two of these configs as a list in this form as well,
````
monai.bundle.run("", **{
    "config_file": ["./config1.json", "./config2.json"], 
    
})
````
But it doesn't seem to handle overriding keys, although it merges non-conflicting keys. 

**My temporary fix**
To achieve the functionality I needed, I've changed the following,
https://github.com/Project-MONAI/MONAI/blob/86c7ecdc1225529a379e2cd58c23fe53aa79b0d0/monai/bundle/config_parser.py#L223

to 
````{python}
    def update(self, pairs: dict[str, Any]):
        """
        Set the ``id`` and the corresponding config content in pairs, see also :py:meth:`__setitem__`.
        For example, ``parser.update({"train#epoch": 100, "train#lr": 0.02})``

        Args:
            pairs: dictionary of `id` and config pairs.

        """
        for k, v in pairs.items():
            if isinstance(v, dict):
                self.update({f'{k}#{_k}': _v for _k, _v in v.items()})
                return 
                
            self[k] = v
````

**Additional context**
Ideally, it would be nice to have the ability to merge configs in the `config_file` list. Is there a way to do this, and am potentially missing something?

### LINKED ISSUE #6387: APIs for combining file-based and dict-based configs
**Is your feature request related to a problem? Please describe.**

the current API is private

https://github.com/Project-MONAI/MONAI/blob/e18097d0a8ba323ee66aeb7830d296bf54f44bdd/monai/bundle/scripts.py#L68-L71

should be made available properly for reading multiple config files and dictionary-based overrides.

### PR DESCRIPTION
**Pull request by wyli**
Fixes #6387
Fixes #5899

### Description
- add api for update_kwargs
- add support of merging multiple configs files and dictionaries
- remove warning message of directory in `runtests.sh`

### Types of changes

- [x] Non-breaking change (fix or new feature that would not break existing functionality).
- [ ] Breaking change (fix or new feature that would cause existing functionality to change).
- [ ] New tests added to cover the changes.
- [ ] Integration tests passed locally by running `./runtests.sh -f -u --net --coverage`.
- [ ] Quick tests passed locally by running `./runtests.sh --quick --unittests  --disttests`.
- [ ] In-line docstrings updated.
- [ ] Documentation updated, tested `make html` command in the `docs/` folder.

### CODE REVIEW

File: monai/bundle/scripts.py [hunk 1]
```diff
@@ -66,28 +66,35 @@
 PPRINT_CONFIG_N = 5
 
 
-def _update_args(args: str | dict | None = None, ignore_none: bool = True, **kwargs: Any) -> dict:
+def update_kwargs(args: str | dict | None = None, ignore_none: bool = True, **kwargs: Any) -> dict:
```
Comment (KumoLiu) [inline monai/bundle/scripts.py:69]: Do we need to support sequence str here and use `load_config_files` in L89?

**Review by ericspod — APPROVED** [review]
LGTM

File: monai/bundle/scripts.py (hunk 1 above)
Comment (wyli) [reply monai/bundle/scripts.py:69]: good point, that logic in in the config parser https://github.com/Project-MONAI/MONAI/blob/fc1350aae1de702404189f334ac17fd6f51321d0/monai/bundle/config_parser.py#L318-L320

but that reminds me there's a case of comma separated filenames https://github.com/Project-MONAI/research-contributions/blob/b2809b5adfa49b7e9220f1e2d1ff26a16f23dc1b/auto3dseg/algorithm_templates/swinunetr/scripts/train.py#L142-L143 I'll update the load config to fix it

Comment (wyli) [reply monai/bundle/scripts.py:69]: had another look and add support for sequence str, which can close https://github.com/Project-MONAI/MONAI/issues/5899 at the CLI level.

**Pull request merged**